lib/identity/partialClient.js

```diff
@@ -54,25 +54,36 @@ class PartialClient extends CryptoConsumer {
           expiry_seconds: 1,
         },
       }
-      const request = await anonAuth.tsv1Fetch(
+      const response = await anonAuth.tsv1Fetch(
         `${this.config.apiUrl}/v2/storage/notes/challenge?id_string=${noteName}`,
         {
           method: 'PATCH',
           'Content-Type': 'application/json',
           body: JSON.stringify(body),
         }
       )
-      await checkStatus(request)
-      return true
-    } catch (_) {
-      return false
+      await checkStatus(response)
+      return {success: true}
+    } catch (e) {
+      let ret = {success: false}
```

Comment:
on second glance, there should also be a default message that says something like "something went wrong" as a catchall for any other errors that may happen. otherwise we swallow the error with no indication that it didn't go as planned

the switch-case below can just set the `message` like `ret.message = "Current password is incorrect"`

Reply:
Ah! My beginner Javascript skills are showing

Reply:
Would it be too much to do something like `Something went wrong. ${e}`

Reply:
you'd want `e.message`, i think. dealers choice. i tend to shy away from displaying potentially confusing errors to users 😆